src/hotspot/share/gc/g1/g1IHOPControl.cpp

```diff
@@ -44,32 +44,37 @@ void G1IHOPControl::update_target_occupancy(size_t new_target_occupancy) {
   _target_occupancy = new_target_occupancy;
 }
 
+void G1IHOPControl::report_statistics(G1NewTracer* new_tracer, size_t occupancy) {
+  print_log(occupancy);
+  send_jfr_event(new_tracer, occupancy);
+}
+
 void G1IHOPControl::update_allocation_info(double allocation_time_s, size_t additional_buffer_size) {
   assert(allocation_time_s >= 0.0, "Allocation time must be positive but is %.3f", allocation_time_s);
 
   _last_allocation_time_s = allocation_time_s;
 }
 
-void G1IHOPControl::print() {
+void G1IHOPControl::print_log(size_t occupancy) {
   assert(_target_occupancy > 0, "Target occupancy still not updated yet.");
   size_t cur_conc_mark_start_threshold = get_conc_mark_start_threshold();
   log_debug(gc, ihop)("Basic information (value update), threshold: %zuB (%1.2f), target occupancy: %zuB, current occupancy: %zuB, "
                       "recent allocation size: %zuB, recent allocation duration: %1.2fms, recent old gen allocation rate: %1.2fB/s, recent marking phase length: %1.2fms",
                       cur_conc_mark_start_threshold,
                       percent_of(cur_conc_mark_start_threshold, _target_occupancy),
                       _target_occupancy,
-                      G1CollectedHeap::heap()->used(),
+                      occupancy,
```

Comment:
Not sure if I misunderstood this or not -- this two value `_target_occupancy` vs `occupancy` mean sth quite different, one for whole-heap and the other for non-young regions.

Based on their names, I'd expect them to refer to the same set of regions.